client/src/main/scala/org/apache/celeborn/client/ApplicationHeartbeater.scala

```diff
@@ -96,8 +99,31 @@ class ApplicationHeartbeater(
     }
   }
 
+  private def unregisterApplication(): Unit = {
+    try {
+      // Then unregister Application
+      logInfo(s"Unregister Application when $appId shutdown")
```

Comment:
@RexXiong, could this log add after askSync? Meanwhile, log the status of `ApplicationLostResponse`?